### PR TITLE
Feat/select share content

### PR DESCRIPTION
**Pull request by pitb2022**
https://github.com/thematters/matters-web/issues/4126

### CODE REVIEW

File: src/components/Forms/CommentFormBeta/index.tsx [hunk 1]
```diff
@@ -170,6 +170,16 @@ export const CommentFormBeta: React.FC<CommentFormBetaProps> = ({
...
 
+  useEffect(() => {
+    const $editor = document.querySelector(
+      `#${formId} .ProseMirror`
+    ) as HTMLElement
+
+    if ($editor && defaultContent) {
+      $editor.innerHTML = defaultContent
+    }
+  }, [defaultContent])
```
Comment (robertu7) [inline src/components/Forms/CommentFormBeta/index.tsx:181]: We don't expect to update the editor content if `defaultContent` is for initialize editor. Better to move this logic back to `<TextSelectionPopover>` or expose an API (or event dispatching?).

File: src/components/TextSelectionPopover/index.tsx
[the remark this answers is not in the document]
Comment (pitb2022) [reply src/components/TextSelectionPopover/index.tsx:1]: 试过用tippy，
1. 无法控制tooltip箭头的方向，
2. 交互(interactive: true)需要有Children，不然会报container is null错误

File: src/components/Forms/CommentFormBeta/index.tsx
Comment (robertu7) [inline src/components/Forms/CommentFormBeta/index.tsx:1]: https://github.com/thematters/matters-web/blob/c0d12c12d8ea259f0e9d13ba4048c138ea9238f7/src/components/Forms/CommentFormBeta/index.tsx#L152-L159 can use `editor` API instead as this PR

File: src/components/Editor/Comment/index.tsx [hunk 2]
```diff
@@ -9,9 +14,15 @@ interface Props {
...
   placeholder?: string
+  setEditor?: (editor: Editor | null) => void
 }
 
-const CommentEditor: React.FC<Props> = ({ content, update, placeholder }) => {
+const CommentEditor: React.FC<Props> = ({
+  content,
+  update,
+  placeholder,
+  setEditor,
```
[the remark this answers is not in the document]
Comment (robertu7) [reply src/components/Editor/Comment/index.tsx:24]: can we use event dispatch instead?

File: src/views/ArticleDetail/index.tsx [hunk 3]
```diff
@@ -157,6 +162,36 @@ const BaseArticleDetail = ({
...
+  }
+  useEventListener(
+    OPEN_COMMENT_LIST_DRAWER,
+    (payload: { [key: string]: any }) => {
+      setCommentDrawerStep('commentList')
+      setQuoteComment(payload.quoteComment)
+      setIsOpenComment(true)
+    }
+  )
+  useEffect(() => {
```
Comment (robertu7) [inline src/views/ArticleDetail/index.tsx:179]: can call editor API directly on `TextSelectionPopover`?

**Review by robertu7 — APPROVED** [review]

**Pull request merged**